OnlineGameStore.Infrastructure/Repositories/Implementations/RepositoryBase.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.EntityFrameworkCore;
+using OnlineGameStore.Infrastructure.Repositories.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OnlineGameStore.Infrastructure.Repositories.Implementations
+{
+    public abstract class RepositoryBase<TEntity> : IRepository<TEntity> where TEntity : class
+    {
+        private readonly DbContext _forumContext;
+
+        protected RepositoryBase(DbContext context)
+        {
+            _forumContext = context;
+        }
+
+        public virtual async Task<IEnumerable<TEntity>> GetAllAsync()
+        {
+            return await _forumContext.Set<TEntity>().ToListAsync();
+        }
+
+        public virtual async Task<TEntity?> GetByIdAsync(Guid id)
+        {
+            return await _forumContext.Set<TEntity>()
+                .FindAsync(id);
+        }
+
+        public virtual async Task<TEntity> AddAsync(TEntity entity)
+        {
+            _forumContext.Set<TEntity>().Add(entity);
+            await _forumContext.SaveChangesAsync();
```

Comment:
So you using save method in unit of work, but doublecalling it here?

Reply:
Fixed